src/dotnet/commands/dotnet-build/CompileContext.cs

```diff
@@ -91,7 +91,7 @@ private bool NeedsRebuilding(ProjectContext project, ProjectDependenciesFacade d
 
         private bool NeedsRebuilding(ProjectContext project, ProjectDependenciesFacade dependencies, string baseBuildPath)
         {
-            var compilerIO = GetCompileIO(project, _args.ConfigValue, baseBuildPath, _args.OutputValue, dependencies);
+            var compilerIO = GetCompileIO(project, _args.ConfigValue, baseBuildPath, _args.OutputValue, dependencies, project == _rootProject);
```

Comment:
Couldn't you make the check in `GetCompileIO` instead of passing it in? Also are `ProjectContext` instances guaranteed to be unique for a single build (i.e. could you have two instances for the same project)?